assets/src/dashboard/animations/animator/output.js

```diff
@@ -19,21 +19,45 @@
  */
 import PropTypes from 'prop-types';
 
-function WithAnimation({ id, style, children }) {
-  return (
-    <div id={id} style={style}>
+/**
+ * Internal dependencies
+ */
+import { ANIMATION_TYPE } from '../constants';
+
+function WithAnimation({
+  id,
+  animationType,
+  animationStyle,
+  containerStyle,
+  children,
+}) {
+  return animationType === ANIMATION_TYPE.REVEAL ? (
+    <div style={{ overflow: 'hidden', ...containerStyle }}>
+      <div id={id} style={{ width: '100%', height: '100%', ...animationStyle }}>
+        {children}
```

Comment:
Overall nesting is a difficult topic. Do you have any top-level thoughts on this? One situation is wrappers. But another one is animation of the same properties. E.g. one effect could animation `transform: translate` and another `transform:scale`. Thus they'd both collide on `transform` and need to be split onto two different divs.

Reply:
Yes, I've been thinking of the nesting problem a bit too and here is what I've come up with, please let me know if you see problems with it or if you simply have a better way:

Let's say you have an `Image` with 4 animation applied to it.  I imagine the data structure for that image to be something like this:
```
{
  pages: [
    {
      type: "image"
      ...
      animations: [
        { type: 'MoveDown', ... },
        { type: 'Bounce', ... },
        { type: 'Spin', ... },
        { type: 'FadeIn', ... }
      ]
    }
  ]
}
```
And then this would generate something like this:
```
<amp-story>
  ...
  <WithAnimation type="MoveDown">
    <WithAnimation type="Bounce">
      <WithAnimation type="Spin">
        <WithAnimation type="FadeIn">
          <Image />
        </WithAnimation>
      </WithAnimation>
    </WithAnimation>
  </WithAnimation>
  ...
</amp-story>
```

Doing this would create quite a lot of nesting (for complex animations), but it'd allow each animation to run on top of each other without any collisions on transform or anything else (although order would be important or results could be funky).  And we could create this with a pretty simple loop of the data structure.

Now, if we wanted to get more fancy, we could loop through all the keyframe data for each applied animation and somehow merge them together (taking into account duration, delay, etc...) and generate a single combined set of animation data that could be applied with just one `WithAnimation` component.  

But even if we preferred the fancy technique in the long run, I feel we could start with brute force nesting and then update the logic to combine keyframes properly at a later date (since the underlying animation data will be the same regardless).

What do you think?

Reply:
Yeah. I think for now this is what we have to do: a div per a set of keyframes. We might be able to get rid of it later either via optimization (reducing number of levels that do not conflict) or via compound animations (assuming that will ever happen).